Merge testing into dev

jessemapel opened the pull request:
Tests passed last night for the testing branch, so this is ready to go.

celbal committed:
Updated the cmake version to 3.10

celbal committed:
Adding configurations for gtest

celbal committed:
Tweaking things for gtest

celbal committed:
Got gtest working and made a small example test

celbal committed:
Updated the cmake version to be 3.10 or greater

krlberry committed:
Merge pull request #527 from makaylas/testing

Gtest and a small example test

krlberry committed:
Added initial Angle test refactor to use gtest

krlberry committed:
Add IException and QDebug tests

krlberry committed:
Added the rest of the IException tests

celbal committed:
Added test file discovery support and seperated the main into another file

Commit:
Merge pull request #534 from makaylas/testing

Added test file discovery support and seperated the main into another…

celbal committed:
Added more tests for FileName

celbal committed:
Updated with the new testing guidelines and added new tests.

Commit:
Added Spectel unittest for gtest environment

celbal committed:
Added gmock

Commit:
Merge pull request #535 from makaylas/testing

Added gmock, more tests for FileName

Commit:
Forgot to save a file

Commit:
Merge conflict

celbal committed:
Merge pull request #540 from USGS-Astrogeology/dev

Update testing from dev

kaitlyndlee committed:
Added Endian unit test.

Commit:
PixelTest (#536)

* initial Pixel test

* Add more test cases for Pixel

* testing parameterization in Pixel test

* testing function parameterization

* parameterize static vs object methods

* basic PixelTests implementation

* clean up and more static tests

* update PixelTests with float/dbl expects

krlberry committed:
Angle test refactor to use gtest (#529)

* Added initial Angle test refactor to use gtest

* Add IException and QDebug tests

* Added the rest of the IException tests

* Update Angle to make private methods public and add tests

* Made changes for review comments

* Cleanup and addition of streaming the actual errors thrown to the EXPECT_TRUE macro

krlberry committed:
Merge branch 'testing' into kberryUSGSgtest

krlberry committed:
Merge pull request #541 from USGS-Astrogeology/kberryUSGSgtest

Add Spectel gTest

celbal committed:
FileName gtest conversion (#568)

* Made two methods private as they are only used within FileName

* Added more tests

* Added more tests

* Update FileNameTests.cpp

jessemapel committed:
BundleSettings unit test (#538)

* added build to git ignore

* Fixed duplicate main in FileNameTests

* Added parameterized test example

* Added more tests

* Mored BundleSettings tests

* Even more tests

* Added mocked targetbody settings for BundleSettings test

* Added BundleSettings save tests

* Fixed BundleSetting save tests to just use a QString

* Added default and copy constructor tests and asignment

SgStapleton commented on isis/CMakeLists.txt:
```diff
@@ -28,15 +30,15 @@ set(PACKAGE            "ISIS")
 
 # Version number
-set(VERSION            "3.5.00.0")
+set(VERSION            "3.6.00.0")
```
Does this version number not look a little off...?

jessemapel replied:
We're on 3.6.0 now right? This just has a bunch of extra 0s for minor versions.

jessemapel replied:
Ian and Makayla updated this when they put in gtest.

jessemapel replied:
In the future we should make these kinds of changes on dev and then merge them into feature branches.

SgStapleton commented on isis/CMakeLists.txt:
```diff
@@ -28,15 +30,15 @@ set(PACKAGE            "ISIS")
-set(VERSION_DATE              "2017-04-24")
-set(THIRD_PARTY_LIBS_VERSION  "v007")
+set(VERSION_DATE              "2018-11-09")
+set(THIRD_PARTY_LIBS_VERSION  "v008")
```
Why did we need this line in here exactly...? (It does not currently exist in dev in any form...)

jessemapel replied:
I am not sure. I am going to look into why this change was made.

jessemapel replied:
Ian and Makayla updated this when they put in gtest.

jessemapel committed:
Update FileName test and merge dev into testing (#598)

* Update CMakeLists.txt with version 3.6.0 (patch 1)

* Update cmake/CMakeLists.txt library version

* Updated the Installation instructions based on user feedback.  Also fixed a problem with the isis3VarInit.py script.

* Updates made in accordance with requests made on this PR by other developers.

* Fix tgo cassis unit test correctly

* Fix tgo cassis unit test correctly (#544)

* Fixed two bug with AutoReg. One occurs with one dimensional pattern cubes and the other causes the revious registration sample/line to be returned on sub-pixel registration failure.

* Fix tgo cassis camera truth again (#554)

* Update docsys Makefile to support wwwdoc (#559)

The documentation is installed under `docs`, not `doc`, so this file has been updated.
`setisis isis3.6.0` and running `make wwwdoc` in `$ISISROOT/../isis/src/docsys` will properly sync the documentation to our web server.

* Update isis3VarInit.py

Unnecessary 'touch'ing, the '>' redirect will create the file if it doesn't exist.
Also, the first redirect should be the single '>'.  Otherwise, multiple runs of this program will continue appending to these files.  They still work, its just a lot of setting and unsetting for no good reason.

* Fixed isis3Startup scripts

* Removed unneeded LineEquation include

* Make this program more pythonic (#589)

* Updated documentation, user interaction, and made the file and directory
handling more 'pythonic'.

* Correcting for PR comments.

* Fixed broken links in the Installation instructions due to the hyperlinks to our GitHub wiki changing.

* Removed unused Parabola class

* Removed old license, added new unlicense (#594)

* Tweaked FileName test to use ISISROOT variable

krlberry commented on isis/CMakeLists.txt:
```diff
@@ -5,7 +5,7 @@
 # Specify the required version of CMake.  If your machine does not
 #  have this, it should be easy to build from https://cmake.org/download/
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.10)
```
Do we need to update anything in the system / default anaconda library to support this?

jessemapel replied:
The new environment files should be used to update the anaconda environments. We should be able to do `conda env update -f environment.yml` and it will update the environment.

krlberry replied:
:+1: Yes -- this isn't a barrier to merging it in so much as a "when this gets merged in, what else may need to happen?"

jessemapel replied:
We also need to add the submodule initialization to the build/clone instructions and the build scripts.

krlberry replied:
@jessemapel :+1: Good point!

SgStapleton replied:
@jessemapel @kberryUSGS Did these get done?

Commit:
App to callable function conversion rough draft (#530)

* adjusted Process to support lambdas with captures, updated crop app

* now compiling with colocated funcs and apps and duplicated symbols accross apps removed

* removed debug prints in cmake scripts

* changes as requested from review

* hapke comment

kaitlyndlee committed:
Added Distance unit tests (#585)

* Added Distance unit tests.

* Changed EXPECT_FLOAT_EQ to EXPECT_DOUBLE_EQ and changed Solar Radii test to use one solar radii.

* Changed unneeded EXPECT_DOUBLE_EQ to EXPECT_EQ.

* Fixed toString test.

SgStapleton committed:
Adding gtests for Color and Matrix (#532)

* Adding gtests for Color and Matrix

* Updating gtest for Matrix

* More work to ColorTests.cpp and MatrixTests.cpp

* Added test for constructor making use of TNT:Array2D in MatrixTests

* Update isis/tests/MatrixTests.cpp

Co-Authored-By: SgStapleton <sgstapleton@usgs.gov>

* Update isis/tests/MatrixTests.cpp

Co-Authored-By: SgStapleton <sgstapleton@usgs.gov>

* Making recommended changes

* Fixing odd GitHub behavior...

There were suggested changes made in PR that were not playing nicely. Could not push correct version for whatever reason.

jessemapel committed:
Add Utilities, clean up testing (#604)

* Update CMakeLists.txt with version 3.6.0 (patch 1)

* Update cmake/CMakeLists.txt library version

* Updated the Installation instructions based on user feedback.  Also fixed a problem with the isis3VarInit.py script.

* Updates made in accordance with requests made on this PR by other developers.

* Fix tgo cassis unit test correctly

* Fix tgo cassis unit test correctly (#544)

* Fixed two bug with AutoReg. One occurs with one dimensional pattern cubes and the other causes the revious registration sample/line to be returned on sub-pixel registration failure.

* Fix tgo cassis camera truth again (#554)

* Update docsys Makefile to support wwwdoc (#559)

The documentation is installed under `docs`, not `doc`, so this file has been updated.
`setisis isis3.6.0` and running `make wwwdoc` in `$ISISROOT/../isis/src/docsys` will properly sync the documentation to our web server.

* Update isis3VarInit.py

Unnecessary 'touch'ing, the '>' redirect will create the file if it doesn't exist.
Also, the first redirect should be the single '>'.  Otherwise, multiple runs of this program will continue appending to these files.  They still work, its just a lot of setting and unsetting for no good reason.

* Fixed isis3Startup scripts

* Removed unneeded LineEquation include

* Make this program more pythonic (#589)

* Updated documentation, user interaction, and made the file and directory
handling more 'pythonic'.

* Correcting for PR comments.

* Fixed broken links in the Installation instructions due to the hyperlinks to our GitHub wiki changing.

* Removed unused Parabola class

* Removed old license, added new unlicense (#594)

* Fixed issue with the License change (#599)

* Removed old license, added new unlicense

* Fixed cmake looking for license in the wrong place

* Added TestUtilities and used in Color tests

* Fixed an issue where app xmls were not being copied

* Tweaked IException test failure messages

SgStapleton committed:
Updating TestUtilities.h and associated files to account for AssertIException to be broken into AssertIExceptionMessage and AssertIExceptionError

krlberry committed:
Merge pull request #605 from SgStapleton/testing2

Modfying TestUtilities.h and associated files

SgStapleton commented:
Conda environments and nightly scripts have been updated for in-house testing
